OpenSim/Simulation/Model/Model.cpp

```diff
@@ -1645,16 +1670,9 @@ int Model::replaceMarkerSet(const SimTK::State& s, MarkerSet& aMarkerSet)
     // Now add the markers from aMarkerSet whose body names match bodies in the engine.
     for (i = 0; i < aMarkerSet.getSize(); i++)
     {
-        // Eran: we make a *copy* since both _markerSet and aMarkerSet own their elements (so they will delete them)
-        Marker* marker = aMarkerSet.get(i).clone();
-        const string& frameName = marker->getFrameName();
-        if (getFrameSet().contains(frameName))
-        {
-            const OpenSim::PhysicalFrame* frame = dynamic_cast<const PhysicalFrame*>(&getFrameSet().get(frameName));
-            if(frame) marker->changeFrame(*frame);
-            upd_MarkerSet().adoptAndAppend(marker);
-            numAdded++;
-        }
+        Marker* marker = aMarkerSet[i].clone();
+        upd_MarkerSet().adoptAndAppend(marker);
+        ++numAdded;
     }
```

Comment:
No need to `finalizeConnections()` explicitly and the only change from the user perspective will be an exception thrown if a Marker's frame is not present.